actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -16,13 +16,65 @@ def initialize(format)
       super "No renderer defined for format: #{format}"
     end
   end
+  # RFC: Consider documenting or testing MissingRenderer
+  #
+  # See <tt>Responder#api_behavior</tt>
+  # https://github.com/plataformatec/responders/blob/f0e093d01e799d8b67952c013daf8808e9ea02db/lib/action_controller/responder.rb#L209
+  #
+  #   # This is the common behavior for formats associated with APIs, such as :xml and :json.
+  #   def api_behavior
+  #     raise MissingRenderer.new(format) unless has_renderer?
+  #   end
+  #
+  #   # Check whether the necessary Renderer is available
+  #   def has_renderer?
+  #     Renderers::RENDERERS.include?(format)
+  #   end
+
+  # See <tt>Renderers.add_serializer</tt>
+  def self.add_serializer(key, &block)
+    Renderers.add_serializer(key, &block)
+  end
+
+  # See <tt>Renderers.remove_serializer</tt>
+  def self.remove_serializer(key)
+    Renderers.remove_serializer(key)
+  end
+
+  # See <tt>Renderers::MissingRenderer</tt>
+  class MissingSerializer < LoadError
+    def initialize(format)
+      super "No serializer defined for format: #{format}"
+    end
+  end
 
   module Renderers
     extend ActiveSupport::Concern
 
+    # A Set containing renderer names that correspond to available renderer procs.
+    # Default values are <tt>:json</tt>, <tt>:js</tt>, <tt>:xml</tt>.
+    RENDERERS = Set.new
```

Comment:
Follow up comment in https://github.com/rails/rails/pull/21496/files#r54989781, WIP pr being pushed while I think about implementation. (I haven't gotten it 'right', yet.)